CMakeLists.txt

```diff
@@ -331,20 +333,60 @@ endif()
 set(TECA_VERSION_DESCR ${tmp} CACHE STRING "TECA version descriptor")
 message(STATUS "TECA version ${TECA_VERSION_DESCR}")
 
-# do we have access to a checkout of TECA_data?
-set(tmp OFF)
-set(TECA_DATA_ROOT "/path/to/TECA_data"
-    CACHE PATH "Path to TECA_data repository")
-if (EXISTS "${TECA_DATA_ROOT}")
-    set(tmp ON)
-    message(STATUS "TECA_data -- available")
-elseif (REQUIRE_TECA_DATA)
-    message(FATAL_ERROR "TECA_data -- required but not found")
+# configure for TECA_data
+set(TECA_DATA_ROOT CACHE PATH "Path to TECA_data repository")
+if (REQUIRE_TECA_DATA)
+    # do we have access to a checkout of TECA_data?
+    if (TECA_DATA_ROOT)
+        if (EXISTS "${TECA_DATA_ROOT}")
+            set(tmp ON)
+            message(STATUS "TECA_data -- available")
+        else()
+            message(FATAL_ERROR "TECA_data -- required but user-set path not found")
+        endif()
+    else()
+        set(tmp ON)
+        ExternalProject_Add(TECA_data
+            SVN_REPOSITORY svn://svn.code.sf.net/p/teca/TECA_data
+            UPDATE_COMMAND svn update
+            CONFIGURE_COMMAND "" BUILD_COMMAND ""
+            INSTALL_COMMAND "" LOG_DOWNLOAD 1)
+        ExternalProject_Get_property(TECA_data SOURCE_DIR)
+        set(TECA_DATA_ROOT ${SOURCE_DIR})
+        message(STATUS "TECA_data -- populating content at ${TECA_DATA_ROOT}")
+    endif()
 else()
     message(STATUS "TECA_data -- not available")
 endif()
 set(TECA_HAS_DATA ${tmp} CACHE BOOL "TECA_data is present")
 
+# configure for TECA_assets
+set(TECA_ASSETS_ROOT CACHE PATH "Path to TECA_assets repository")
+if (REQUIRE_TECA_ASSETS)
+    # do we have access to a checkout of TECA_assets?
+    if (TECA_ASSETS_ROOT)
+        if (EXISTS "${TECA_ASSETS_ROOT}")
+            set(tmp ON)
+            message(STATUS "TECA_assets -- available")
+        else()
+            message(FATAL_ERROR "TECA_assets -- required but user-set path not found")
+        endif()
+    else()
+        set(tmp ON)
+        ExternalProject_Add(TECA_assets
+            SVN_REPOSITORY svn://svn.code.sf.net/p/teca/TECA_assets
+            UPDATE_COMMAND svn update
+            CONFIGURE_COMMAND "" BUILD_COMMAND ""
+            INSTALL_COMMAND "" LOG_DOWNLOAD 1)
+        ExternalProject_Get_property(TECA_assets SOURCE_DIR)
+        set(TECA_ASSETS_ROOT ${SOURCE_DIR})
```

Comment:
I believe that this will make a new variable and will not update the cached variable of the same name declared above. The cmake docs confirm this. This code should be restructured so the second variable is not needed. https://cmake.org/cmake/help/v3.18/manual/cmake-language.7.html#cmake-language-variables